lib/diagnostic.js

```diff
@@ -5,16 +5,24 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (previous, current) {
 
+function arrayMin(arr) {
+  const len = arr.length, min = Infinity;
```

Comment:
Interesting solution, I guess comparing something to `Infinity` at the start works